homework-assignments-1/task2.ts

```diff
@@ -0,0 +1,16 @@
+class User {
+  private username: string;
+  private password: string;
+  constructor(username: string, password: string) {
+    this.username = username;
+    this.password = password;
+  }
+  validateLogin(inputPassword: string): boolean {
```

Comment:
The login validation process should be handled internally by the method without exposing the details of how passwords are compared.

`  public validateLogin(inputPassword: string): boolean {
    // Abstracts the logic of password comparison
    return this.checkPassword(inputPassword);
  }

  // Private method for password comparison
  private checkPassword(inputPassword: string): boolean {
    // Compare the input password with the actual password
    return inputPassword === this.password;
  }`